pkg/bloomgateway/worker.go

```diff
@@ -210,7 +210,9 @@ func (w *worker) stopping(err error) error {
 }
 
 func (w *worker) processBlocksWithCallback(taskCtx context.Context, tenant string, day time.Time, blockRefs []bloomshipper.BlockRef, boundedRefs []boundedTasks) error {
+	storeFetchStart := time.Now()
 	return w.store.ForEach(taskCtx, tenant, blockRefs, func(bq *v1.BlockQuerier, minFp, maxFp uint64) error {
+		w.metrics.storeAccessLatency.WithLabelValues(w.id, "ForEach").Observe(time.Since(storeFetchStart).Seconds())
```

Comment:
I believe this metric includes not only the time it took to download the block, but also the time it took to process all the previous blocks.
let's say we have 10 blocks, then when we process block #10, `storeFetchStart` will be the time when we just started the whole function. 
is it expected?

Reply:
That's right. Needs to be changed.